targets/feather-stm32f405.json

```diff
@@ -0,0 +1,16 @@
+{
+  "inherits": ["cortex-m4"],
+  "build-tags": ["feather_stm32f405", "stm32f405", "stm32f4", "stm32"],
+  "automatic-stack-size": false,
+  "default-stack-size": 1024,
+  "cflags": [
+    "-Qunused-arguments"
+  ],
+  "linkerscript": "targets/stm32f405.ld",
+  "extra-files": [
+    "src/device/stm32/stm32f405.s"
+  ],
+  "openocd-transport": "swd",
+  "openocd-interface": "jlink",
+  "openocd-target": "stm32f4x"
```

Comment:
I think the default writing method should be dfu-util.

I added flash-command and confirmed that it can flash by dfu-util.
The reason you can''t flash properly originally is because it was {hex}.

```
{
  "inherits": ["cortex-m4"],
  "build-tags": ["feather_stm32f405", "stm32f405", "stm32f4", "stm32"],
  "automatic-stack-size": false,
  "default-stack-size": 1024,
  "cflags": [
    "-Qunused-arguments"
  ],
  "linkerscript": "targets/stm32f405.ld",
  "extra-files": [
    "src/device/stm32/stm32f405.s"
  ],
  "flash-method": "command",
  "flash-command": "dfu-util -a 0 --dfuse-address 0x08000000 -D {bin}",
  "openocd-transport": "swd",
  "openocd-interface": "jlink",
  "openocd-target": "stm32f4x"
}
```

You can flash in any of the following ways

```
$ tinygo flash -target feather-stm32f405 -size short ./src/examples/blinky1
$ tinygo flash -target feather-stm32f405 -size short -programmer jlink ./src/examples/blinky1
$ tinygo flash -target feather-stm32f405 -size short -programmer cmsis-dap ./src/examples/blinky1
```

Reply:
The following is better.

```
  "flash-command": "dfu-util --alt 0 --dfuse-address 0x08000000 --download {bin}",
```

Reply:
The latest commit confirms it.
Very good.

```
$ tinygo flash --target feather-stm32f405 --size short ./src/examples/b1
   code    data     bss |   flash     ram
   2908       0    4144 |    2908    4144
dfu-util 0.9

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2016 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

dfu-util: Invalid DFU suffix signature
dfu-util: A valid DFU suffix will be required in a future dfu-util release!!!
Opening DFU capable USB device...
ID 0483:df11
Run-time device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Setting #0 ...
Determining device status: state = dfuERROR, status = 10
dfuERROR, clearing status
Determining device status: state = dfuIDLE, status = 0
dfuIDLE, continuing
DFU mode device DFU version 011a
Device returned transfer size 2048
DfuSe interface name: "Internal Flash  "
Downloading to address = 0x08000000, size = 2908
Download	[=========================] 100%         2908 bytes
Download done.
File downloaded successfully
```

```
$ tinygo flash --target feather-stm32f405 --size short --programmer cmsis-dap ./src/examples/b5
   code    data     bss |   flash     ram
   2908       0    4144 |    2908    4144
Open On-Chip Debugger 0.10.0+dev-g5c9350d2 (2020-08-07-12:56)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
swd
Info : Using CMSIS-DAPv2 interface with VID:PID=0x6666:0x6666, serial=123456
Info : CMSIS-DAP: SWD  Supported
Info : CMSIS-DAP: FW Version = v0.1
Info : CMSIS-DAP: Serial# = 123456
Info : CMSIS-DAP: Interface Initialised (SWD)
Info : SWCLK/TCK = 1 SWDIO/TMS = 1 TDI = 1 TDO = 1 nTRST = 1 nRESET = 1
Info : CMSIS-DAP: Interface ready
Info : clock speed 2000 kHz
Info : SWD DPIDR 0x2ba01477
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : starting gdb server for stm32f4x.cpu on 3333
Info : Listening on port 3333 for gdb connections
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x0800059a msp: 0x20001000
Info : High speed (adapter speed 8000) may be limited by adapter firmware.
** Programming Started **
Info : device id = 0x10076413
Info : flash size = 1024 kbytes
** Programming Finished **
** Verify Started **
** Verified OK **
** Resetting Target **
shutdown command invoked

```